ci_tools/list_docs_tovalidate.py

```diff
@@ -0,0 +1,53 @@
+""" Script to list the objects that should be numpydoc validated
+"""
+import ast
+from _ast import FunctionDef, ClassDef, Module
+import argparse
+from git_evaluation_tools import get_diff_as_json as gdj
+
+parser = argparse.ArgumentParser(description='List the objects with docstrings in the files provided')
+parser.add_argument('gitdiff', metavar='gitdiff', type=str,
+                        help='Output of git diff between PR branch and base branch')
+parser.add_argument('output', metavar='output', type=str,
+                        help='File to save the output to')
+args = parser.parse_args()
+
+results = gdj(args.gitdiff)
+changes = {}
+for file, upds in results.items():
+    if file.startswith('pyccel/'):
```

Comment:
It is not necessary here as we know that this file will be run on linux but it is good practice to use the [`pathlib`](https://docs.python.org/3/library/pathlib.html) module (or the [`os.path`](https://docs.python.org/3/library/os.path.html#module-os.path) module) to handle anything file related.
For example on Windows you would never enter this if as the file would start with `pyccel\\`

So here you could have:
```python
if pathlib.PurePath(file).parents[0] == 'pyccel':
```

and below you could have:
```python
prefix = '.'.join(pathlib.PurePath(file).with_suffix('').parts)
```

Reply:
obviously you can also do `from pathlib import PurePath` for shorter commands

Reply:
Sorry I think this should be `parts` not `parents`

Reply:
True, I will fix it!